tests/components/sensibo/test_coordinator.py

```diff
@@ -0,0 +1,70 @@
+"""The test for the sensibo coordinator."""
+from __future__ import annotations
+
+from datetime import timedelta
+from unittest.mock import patch
+
+from pysensibo.exceptions import AuthenticationError, SensiboError
+from pysensibo.model import SensiboData
+import pytest
+
+from homeassistant.components.sensibo.const import DOMAIN
+from homeassistant.components.sensibo.coordinator import SensiboDataUpdateCoordinator
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import UpdateFailed
+from homeassistant.util import dt
+
+from tests.common import async_fire_time_changed
+
+
+async def test_coordinator(hass: HomeAssistant, load_int: ConfigEntry) -> None:
+    """Test the Sensibo coordinator."""
+
+    coordinator: SensiboDataUpdateCoordinator = hass.data[DOMAIN][load_int.entry_id]
+    assert coordinator.data.parsed["ABC999111"].state == "heat"
+
+
+async def test_coordinator_errors(hass: HomeAssistant, load_int: ConfigEntry) -> None:
+    """Test the Sensibo coordinator errors."""
+
+    coordinator: SensiboDataUpdateCoordinator = hass.data[DOMAIN][load_int.entry_id]
+    coordinator.data.parsed["ABC999111"].state = "heat"
+
+    assert coordinator.last_update_success is True
+
+    with patch(
+        "homeassistant.components.sensibo.coordinator.SensiboClient.async_get_devices_data",
+        side_effect=AuthenticationError,
+    ):
+        with pytest.raises(ConfigEntryAuthFailed):
+            await coordinator._async_update_data()  # pylint: disable=protected-access
+
+    with patch(
+        "homeassistant.components.sensibo.coordinator.SensiboClient.async_get_devices_data",
+        side_effect=SensiboError,
+    ):
+        with pytest.raises(UpdateFailed):
+            await coordinator._async_update_data()  # pylint: disable=protected-access
```

Comment:
I think it would be better change time to cause an update:
https://github.com/home-assistant/core/blob/1f6e934a8eeef066091597e3e9a8c117fb58a5af/tests/components/sensibo/test_binary_sensor.py#L38

Reply:
I agree but would not catch the errors then as the time-change itself does not catch it.
Therefore I added an assert in the very end to check that an update was unsuccessful with such conditions.